zulip_bots/zulip_bots/bots/baremetrics/baremetrics.py

```diff
@@ -20,6 +21,19 @@ def initialize(self, bot_handler: Any) -> None:
                              'List the sources', 'List the plans for the source', 'List the customers in the source',
                              'List the subscriptions in the source']
 
+        self.check_api_key()
+
+    def check_api_key(self) -> None:
+        url = "https://api.baremetrics.com/v1/account"
+        test_query_response = requests.get(url, headers=self.auth_header)
+        test_query_data = test_query_response.json()
+
+        try:
+            if test_query_data['error'] == "Unauthorized. Token not found":
+                logging.error('API Key not valid. Please see doc.md to find out how to get it.')
```

Comment:
You can add a sys.exit(1) here, since this shouldn't cause any problems as long as we do not cover check_access_token in tests.